test/presubmit_test.go

```diff
@@ -75,3 +78,133 @@ func hasHyphen(path string, info os.FileInfo, _ error) error {
 
 	return nil
 }
+
+// Test if error messages start with an upper case.
+func TestLowercaseLog(t *testing.T) {
+	err := filepath.Walk("..", hasLowercase)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func hasLowercase(path string, info os.FileInfo, _ error) error {
+	// only for regular files, not starting with a . and those that are go files.
+	if !info.Mode().IsRegular() {
+		return nil
+	}
+	if strings.HasPrefix(path, "../.") {
+		return nil
+	}
+	if !strings.HasSuffix(path, "_test.go") {
+		return nil
+	}
+
+	fs := token.NewFileSet()
+	f, err := parser.ParseFile(fs, path, nil, parser.AllErrors)
+	if err != nil {
+		return err
+	}
+	l := &logfmt{}
+	ast.Walk(l, f)
+	if l.err != nil {
+		return l.err
+	}
+	return nil
+}
+
+type logfmt struct {
+	err error
+}
+
+func (l logfmt) Visit(n ast.Node) ast.Visitor {
+	if n == nil {
+		return nil
+	}
+	ce, ok := n.(*ast.CallExpr)
+	if !ok {
+		return l
+	}
+	se, ok := ce.Fun.(*ast.SelectorExpr)
+	if !ok {
+		return l
+	}
+	id, ok := se.X.(*ast.Ident)
+	if !ok {
+		return l
+	}
+	if id.Name != "t" { //t *testing.T
```

Comment:
Is that comment intentional? Just trying to understand the specifics here.

Reply:
yes, "t" signals that Errorf is called from a var named `t` which we assume is a *testing.T (although the type isn't checked)

Reply:
Thanks for the context. Not sure I like that unchecked assumption, but ok with just the comment.